src/main/java/cloud/eppo/ConfigurationStore.java

```diff
@@ -1,53 +1,106 @@
-  // TODO: another map of bandit flag variations to handle no action case
-  static ConfigurationStore instance = null;
+  private Map<String, FlagConfig> flags;
+  private Map<String, BanditReference> banditReferences;
+  private Map<String, BanditParameters> banditParameters;
 
   public ConfigurationStore() {
-    // TODO: handle caching
+    flags = new ConcurrentHashMap<>();
+    banditReferences = new ConcurrentHashMap<>();
+    banditParameters = new ConcurrentHashMap<>();
   }
 
-  public void setFlagsFromJsonString(String jsonString) throws JsonProcessingException {
-    FlagConfigResponse config = mapper.readValue(jsonString, FlagConfigResponse.class);
+  public void setFlagsFromJsonString(String jsonString) {
+    FlagConfigResponse config;
+
+    try {
+      config = mapper.readValue(jsonString, FlagConfigResponse.class);
+    } catch (JsonProcessingException e) {
+      log.error("Unable to parse flag configuration response");
+      throw new RuntimeException(e);
+    }
+
     if (config == null || config.getFlags() == null) {
       log.warn("Flags missing in configuration response");
       flags = new ConcurrentHashMap<>();
+      banditReferences = new ConcurrentHashMap<>();
     } else {
       // TODO: atomic flags to prevent clobbering like android does
       // Record that flags were set from a response so we don't later clobber them with a
       // slow cache read
       flags = new ConcurrentHashMap<>(config.getFlags());
-      log.debug("Loaded " + flags.size() + " flags from configuration response");
+      banditReferences = new ConcurrentHashMap<>(config.getBanditReferences());
+      log.debug("Loaded {} flags from configuration response", flags.size());
     }
   }
 
-  public BanditParameters getBanditParameters(String banditKey) {
-    return this.banditParameters.get(banditKey);
-  }
-
   public FlagConfig getFlag(String flagKey) {
     if (flags == null) {
-      log.warn("Request for flag " + flagKey + " before flags have been loaded");
+      log.warn("Request for flag {} before flags have been loaded", flagKey);
       return null;
     } else if (flags.isEmpty()) {
-      log.warn("Request for flag " + flagKey + " with empty flags");
+      log.warn("Request for flag {} with empty flags", flagKey);
     }
     return flags.get(flagKey);
   }
+
+  public String banditKeyForVariation(String flagKey, String variationValue) {
+    // Note: In practice this double loop should be quite quick as the number of bandits and bandit
+    // variations will be small
+    //       Should this ever change, we can optimize things.
+    for (Map.Entry<String, BanditReference> banditEntry : banditReferences.entrySet()) {
+      BanditReference banditReference = banditEntry.getValue();
+      for (BanditFlagVariation banditFlagVariation : banditReference.getFlagVariations()) {
+        if (banditFlagVariation.getFlagKey().equals(flagKey)
```

Comment:
While we _could_ use the shortcut of assuming the variation key and value and bandit key are all the same, since the UFC explicitly provides the information to be more specific, I'll take advantage of it. We can have other SDKs do this once needed.